pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,335 @@
+	// List of environment variable to set in the container
+	Env []KeyValue
+	// Mount specifies mounts for the container
+	Mounts []Mount
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Privileged runs the container in the privileged mode.
+	Privileged bool
+	// ReadOnlyRootFS sets the root filesystem of the container to be
+	// read-only.
+	ReadOnlyRootFS bool
+	// Path to store the container log on the host (i.e., outside of the
+	// sandbox).
+	LogPath string
+
+	// Linux contains configuration specific to Linux containers.
+	Linux *LinuxContainerConfig
+}
+
+// LinuxContainerConfig contains platform-specific configuration for
+// Linux-based containers.
+type LinuxContainerConfig struct {
+	// Resources specification for the container.
+	Resources LinuxContainerResources
+	// AddCapabilities lists capabilities to add.
+	AddCapabilities []string
+	// DropCapabilities lists capabilities to drop.
+	DropCapabilities []string
+	// SELinux is the SELinux context to be applied.
+	SELinux *api.SELinuxOptions
+}
+
+// LinuxContainerResources specifies Linux specific configuration for
+// resources.
+// TODO: Consider using Resources from opencontainers/runtime-spec/specs-go
+// directly.
+type LinuxContainerResources struct {
+	// CPU CFS (Completely Fair Scheduler) period
+	CPUPeriod int64
+	// CPU CFS (Completely Fair Scheduler) quota
+	CPUQuota int64
+	// CPU shares (relative weight vs. other containers)
+	CPUShares int64
+	// Memory limit in bytes
+	MemoryLimitInBytes int64
+}
+
+// ContainerFilter is used to filter containers.
+type ContainerFilter struct {
+	// Name of the container.
+	Name string
+	// ID of the container.
+	ID string
+	// State of the contianer.
+	State ContainerState
+	// LabelSelector to select matches.
+	LabelSelector unversioned.LabelSelector
+}
+
+type StreamOptions struct {
+	TTY          bool
+	InputStream  io.Reader
+	OutputStream io.Writer
+	ErrorStream  io.Writer
+}
+
+// KeyValue represents a key-value pair.
+type KeyValue struct {
+	Key   string
+	Value string
+}
+
+type ContainerMetricsGetter interface {
+	// Metrics returns the stats of the container.
+	// TODO: Reuse cadvisor's api for now. Should re-evaluate whether we need
+	// all the fields.
+	Metrics(id string) (cadvisorapiv2.ContainerInfo, error)
+}
+
+// ImageOperations offers basic image operations.
+type ImageOperations interface {
+	// List lists the existing images.
+	List() ([]Image, error)
+	// Pull pulls an image with authentication config.
+	Pull(image ImageSpec, auth AuthConfig) error
+	// Remove removes an image.
+	Remove(image ImageSpec) error
+	// Status returns the status of an image.
+	Status(image ImageSpec) (Image, error)
+}
+
+// ImageMetricsGetter returns metrics about images.
+type ImageMetricsGetter interface {
```

Comment:
I didn't merge this with `ImageOperations` because currently we get this information from cadvisor

/cc @timstclair